pkgs/by-name/ht/hterm/package.nix

```diff
@@ -0,0 +1,77 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  cairo,
+  pango,
+  libpng,
+  expat,
+  fontconfig,
+  gtk2,
+  xorg,
+  autoPatchelfHook,
+  copyDesktopItems,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hterm";
+  version = "0.8.9";
+
+  src =
+    let
+      versionWithoutDots = builtins.replaceStrings [ "." ] [ "" ] finalAttrs.version;
+    in
+    if stdenv.targetPlatform.is64bit then
+      fetchurl {
+        url = "https://www.der-hammer.info/terminal/hterm${versionWithoutDots}-linux-64.tgz";
+        hash = "sha256-DY+X7FaU1UBbNf/Kgy4TzBZiocQ4/TpJW3KLW1iu0M0=";
+      }
+    else
+      fetchurl {
+        url = "https://www.der-hammer.info/terminal/hterm${versionWithoutDots}-linux-32.tgz";
+        hash = "sha256-7wJFCpeXNMX94tk0QVc0T22cbv3ODIswFge5Cs0JhI8=";
+      };
+  sourceRoot = ".";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    cairo
+    pango
+    libpng
+    expat
+    fontconfig.lib
+    gtk2
+    xorg.libSM
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    install -m755 -D hterm $out/bin/hterm
+    install -m644 -D desktop/hterm.png $out/share/pixmaps/hterm.png
+    runHook postInstall
+  '';
+
+  desktopItems = [ "desktop/hterm.desktop" ];
```

Comment:
Install this manually.

Reply:
Thanks for the suggestion, I will install it manually.

However I am wondering why I should not use the copyDesktopItems hook, that was made for the purpose of installing desktop files with the correct permissions to the correct location.

Reply:
Usually, when the upstream project provides a file, we either 
- use it as is 
- patch it, in cases like inexistent paths (say, `/usr/bin/hterm` being cited)
- ignore completely and create our own if the file is utterly broken :)

in this order of preference.

The `copyDesktopItems` hook is intended to be used in conjunction with `makeDesktopItem`, a template that receives inputs, checks them and emit a file in the Nix Store.
Since they are meant to generate the files and deal with corner cases like interactions with other hooks and functions from Nixpkgs, they are naturally more complex.

There is no need to bring all this complexity just to copy a file already provided by upstream.

This would be similar to use rsync to copy a local file to a local directory. Too much complexity with no discernible benefit.